src/ripple/overlay/impl/PeerImp.cpp

```diff
@@ -1965,6 +2009,134 @@ PeerImp::onMessage (std::shared_ptr <protocol::TMHaveTransactionSet> const& m)
+                auto expected = Sanity::unknown;
+                sanity_.compare_exchange_strong(expected, Sanity::sane);
+
+                std::lock_guard<std::mutex> sl(recentLock_);
+
+                assert(applyResult.sequence && applyResult.publisherKey);
+                auto const& pubKey = *applyResult.publisherKey;
+#ifndef NDEBUG
+                if (auto const iter = publisherListSequences_.find(pubKey);
+                    iter != publisherListSequences_.end())
+                {
+                    assert(iter->second < *applyResult.sequence);
+                }
+#endif
+                publisherListSequences_[pubKey] = *applyResult.sequence;
+            }
+            break;
+        case ListDisposition::same_sequence:
+            JLOG (p_journal_.warn()) <<
+                "Validator list with current sequence from peer " <<
+                remote_address_;
+            // Charging this fee here won't hurt the peer in the normal
+            // course of operation (ie. refresh every 5 minutes), but
+            // will add up if the peer is misbehaving.
+            fee_ = Resource::feeUnwantedData;
+#ifndef NDEBUG
+            {
+                std::lock_guard<std::mutex> sl(recentLock_);
+                assert(applyResult.sequence && applyResult.publisherKey);
+                assert(publisherListSequences_[*applyResult.publisherKey]
+                    == *applyResult.sequence);
+            }
+#endif // !NDEBUG
+
+            break;
+        case ListDisposition::stale:
+            JLOG (p_journal_.warn()) <<
+                "Stale validator list from peer " << remote_address_;
+            // There are very few good reasons for a peer to send an
+            // old list, particularly more than once.
+            fee_ = Resource::feeBadData;
+            break;
+        case ListDisposition::untrusted:
+            JLOG (p_journal_.warn()) <<
+                "Untrusted validator list from peer " << remote_address_;
+            // Charging this fee here won't hurt the peer in the normal
+            // course of operation (ie. refresh every 5 minutes), but
+            // will add up if the peer is misbehaving.
+            fee_ = Resource::feeUnwantedData;
+            break;
+        case ListDisposition::invalid:
+            JLOG (p_journal_.warn()) <<
+                "Invalid validator list from peer " << remote_address_;
+            // This shouldn't ever happen with a well-behaved peer
+            fee_ = Resource::feeInvalidSignature;
+            break;
+        case ListDisposition::unsupported_version:
+            JLOG (p_journal_.warn()) <<
+                "Unsupported version validator list from peer " <<
+                remote_address_;
+            // During a version transition, this may be legitimate.
+            // If it happens frequently, that's probably bad.
+            fee_ = Resource::feeBadData;
+            break;
+        default:
+            BOOST_ASSERT(false);
```

Comment:
Is there a point in using `BOOST_ASSERT` instead of good ole' `std::assert`?

Reply:
Nope. Fixed.